### PR TITLE
Basic tracking support

### PR DESCRIPTION
**Pull request by tolysz**
Basic tracking support
Works but needs more work. It can be merged.

### CODE REVIEW

File: src/OpenCV/ImgProc/CascadeClassifier.hs [hunk 1]
```diff
@@ -140,3 +142,80 @@ cascadeClassifierDetectMultiScale cc scaleFactor minNeighbours minSize maxSize s
...
+      } |]
+      numRects <- fromIntegral <$> peek numRectsPtr
+      rectsPtr <- peek rectsPtrPtr
+      rejectLevelsPtr <- peek rejectLevelsPtrPtr
+      levelWeightsPtr <- peek levelWeightsPtrPtr
+      rects :: [Rect Int32] <- peekArray numRects rectsPtr >>= mapM (fromPtr . return)
+      rejectLevels :: [Int32] <- peekArray numRects rejectLevelsPtr -- >>= mapM (fromPtr . return)
+      levelWeights :: [Double] <- map realToFrac <$> peekArray numRects levelWeightsPtr
+
+      traceShow (rejectLevels, levelWeights) $ return ()
```
Comment (basvandijk) [inline src/OpenCV/ImgProc/CascadeClassifier.hs]: This is probably a left-over from debugging. Could you remove it?

File: src/OpenCV/ImgProc/CascadeClassifier.hs [hunk 2]
```diff
@@ -27,6 +28,7 @@ import "this" OpenCV.Internal.C.Inline ( openCvCtx )
 import "this" OpenCV.Internal.C.Types
 import "this" OpenCV.Internal
 import "this" OpenCV.TypeLevel
+import "base" Debug.Trace
```
Comment (basvandijk) [inline src/OpenCV/ImgProc/CascadeClassifier.hs]: This is probably a left-over from debugging. Could you remove it?

**Comment by tolysz** [comment]
I fix the code according to your suggestions, the amend/force push is destroying the review. Maybe the better is to have more commits but then "merge with squash" via github?

**Comment by basvandijk** [comment]
It's no problem to destroy the review. If you've addressed the points then I can review again and merge it.

**Comment by basvandijk** [comment]
Hi @tolysz, I think your forgot to add `opencv-extra/include/tracker.hpp` because I get the build error:

```
src/OpenCV/Extra/Tracking.cpp:6:24: error:
     fatal error: tracking.hpp: No such file or directory
``` 

Could you add it please? Then I'll merge.

**Comment by tolysz** [comment]
Ok, re-added. ;) the git reset is to blame. On the up side, it was enough or re-ties to make travis green!

**Comment by basvandijk** [comment]
Awesome example!

**Pull request merged**